src/libsyntax/attr/builtin.rs

```diff
@@ -945,11 +945,6 @@ pub fn check_builtin_attribute(
     // Some special attributes like `cfg` must be checked
     // before the generic check, so we skip them here.
     let should_skip = |name| name == sym::cfg;
-    // Some of previously accepted forms were used in practice,
-    // report them as warnings for now.
-    let should_warn = |name| name == sym::doc || name == sym::ignore ||
-                             name == sym::inline || name == sym::link ||
-                             name == sym::test || name == sym::bench;
```

Comment:
Too soon, IMO.
This is going directly from a warning to hard error and is going to cause a lot of [indirect fallout](https://github.com/rust-lang/rust/pull/57321#issuecomment-451685517).
I'd rather keep the whitelist (while still making the lint deny-by-default).

Reply:
Done.